packages/trader/src/App/Components/Elements/VideoPlayer/video-controls.tsx

```diff
@@ -0,0 +1,129 @@
+    has_enlarged_dot?: boolean;
+    is_animated?: boolean;
+    is_ended?: boolean;
+    is_playing?: boolean;
+    is_mobile?: boolean;
+    is_muted?: boolean;
+    onRewind: (e: React.MouseEvent<HTMLDivElement>) => void;
+    onVolumeChange: (new_value: number) => void;
+    onPlaybackRateChange: (new_value: number) => void;
+    progress_bar_filled_ref: React.RefObject<HTMLDivElement>;
+    progress_bar_ref: React.RefObject<HTMLDivElement>;
+    progress_dot_ref: React.RefObject<HTMLSpanElement>;
+    show_controls?: boolean;
+    togglePlay: (e: React.MouseEvent<HTMLButtonElement> | React.KeyboardEvent<HTMLButtonElement>) => void;
+    toggleMute: (new_value: boolean) => void;
+    video_duration?: number;
+    volume?: number;
+};
+
+const VideoControls = ({
+    block_controls,
+    current_time,
+    dragStartHandler,
+    has_enlarged_dot,
+    is_animated,
+    is_ended,
+    is_playing,
+    is_mobile,
+    is_muted,
+    onRewind,
+    onVolumeChange,
+    onPlaybackRateChange,
+    progress_bar_filled_ref,
+    progress_bar_ref,
+    progress_dot_ref,
+    show_controls,
+    togglePlay,
+    toggleMute,
+    video_duration,
+    volume,
+}: TVideoControls) => {
+    const [is_drag_dot_visible, setIsDragDotVisible] = React.useState(false);
+
+    return (
+        <div
+            className={classNames('player__controls__wrapper', {
+                'player__controls__wrapper--visible': show_controls,
+                'player__controls__wrapper--interactive': show_controls,
+            })}
+        >
+            <div
+                className='player__controls__progress-bar'
+                onClick={onRewind}
+                onKeyDown={undefined}
+                onMouseOver={() => setIsDragDotVisible(true)}
+                onMouseLeave={() => setIsDragDotVisible(false)}
+                ref={progress_bar_ref}
+            >
+                <div
+                    className={classNames('player__controls__progress-bar__filled', {
+                        'player__controls__progress-bar__filled--animated': is_animated,
+                        'player__controls__progress-bar__filled--ended': is_ended,
+                    })}
+                    ref={progress_bar_filled_ref}
+                >
+                    {(is_mobile || is_drag_dot_visible) && (
+                        <span
+                            className={classNames('player__progress-dot', {
+                                'player__progress-dot--enlarged': has_enlarged_dot,
+                            })}
+                            onMouseDown={dragStartHandler}
+                            onTouchStart={dragStartHandler}
+                            onDragStart={() => false}
+                            ref={progress_dot_ref}
+                        />
+                    )}
+                </div>
+            </div>
+            <div
+                onClick={e => e.stopPropagation()}
```

Comment:
just a question: is this `stopPropagation` needed to avoid adding `stopPropagation` to all callbacks in the elements inside this `div`?
Can `role='button'` help here instread of `onKeyDown={undefined}`?